frontend/src/components/EditCurrentSchedule/ScheduleDialog/ScheduleDialog.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import TextField from '@material-ui/core/TextField';
+import { dialogTypes, dialogCloseButton, dialogChooseButton } from '../../../constants/dialogs';
+import {
+    COMMON_SCHEDULE_DIALOG_TITLE,
+    COMMON_ROOM_IS_UNAVAILABLE,
+    COMMON_AVAILABLE,
+    COMMON_UNAVAILABLE,
+    COMMON_TEACHER_IS_UNAVAILABLE,
+} from '../../../constants/translationLabels/common';
+import { FORM_ROOM_LABEL } from '../../../constants/translationLabels/formElements';
+import './ScheduleDialog.scss';
+import CustomDialog from '../../../containers/Dialogs/CustomDialog';
+import { sortByName } from '../../../helper/sortArray';
+import '../../../share/DialogWindows/dialog.scss';
+import i18n from '../../../i18n';
+
+const ScheduleDialog = (props) => {
+    const {
+        onClose,
+        setOpenConfirmDialog,
+        handleChangeSchedule,
+        itemData,
+        open,
+        rooms,
+        t,
+        availability,
+        isOpenConfirmDialog,
+    } = props;
+
+    const [room, setRoom] = useState('');
+    const [warnings, setWarnings] = useState([]);
+
+    const getOptionLabel = (option) => {
```

Comment:
move to utils, selectUtils.js

Reply:
It will not be possible to transfer to selectUtils because this function is used only in this file

Reply:
done